### PR TITLE
refactor: remove obsolete qt algorithm usage

### PR DESCRIPTION
**Pull request by fanquake**
`qStablesort`, `qSort`, `qLowerBound` and `qUpperBound` have been marked as obsolete since at least Qt 5.9: [Obsolete Members for QtAlgorithms](https://doc.qt.io/qt-5.9/qtalgorithms-obsolete.html). 

This pull request replaces their usage with the suggested `std::` replacements.

This also removes some warning spam when compiling against newer Qt (5.13.0 via brew):
```bash
  CXX      qt/libbitcoinqt_a-walletcontroller.o
qt/transactiontablemodel.cpp:96:52: warning: 'qLowerBound<QList<TransactionRecord>::iterator, uint256, TxLessThan>' is deprecated: Use std::lower_bound [-Wdeprecated-declarations]
        QList<TransactionRecord>::iterator lower = qLowerBound(
qt/transactiontablemodel.cpp:98:52: warning: 'qUpperBound<QList<TransactionRecord>::iterator, uint256, TxLessThan>' is deprecated: Use std::upper_bound [-Wdeprecated-declarations]
        QList<TransactionRecord>::iterator upper = qUpperBound(
```

```bash
  CXX      qt/libbitcoinqt_a-moc_walletcontroller.o
qt/bantablemodel.cpp:64:13: warning: 'qStableSort<QList<CCombinedBan>::iterator, BannedNodeLessThan>' is deprecated: Use std::stable_sort [-Wdeprecated-declarations]
            qStableSort(cachedBanlist.begin(), cachedBanlist.end(), BannedNodeLessThan(sortColumn, sortOrder));
```

```bash
  CXX      qt/libbitcoinqt_a-sendcoinsentry.o
qt/recentrequeststablemodel.cpp:205:5: warning: 'qSort<QList<RecentRequestEntry>::iterator, RecentRequestEntryLessThan>' is deprecated: Use std::sort [-Wdeprecated-declarations]
    qSort(list.begin(), list.end(), RecentRequestEntryLessThan(column, order));
```

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK

File: src/qt/addresstablemodel.cpp [hunk 1]
```diff
@@ -89,7 +89,7 @@ class AddressTablePriv
         // qLowerBound() and qUpperBound() require our cachedAddressTable list to be sorted in asc order
```
Comment (hebasto) [inline src/qt/addresstablemodel.cpp]: Could the comment be fixed by removing/replacement mentions of `qLowerBound()` and `qUpperBound()`?

Comment (laanwj) [reply src/qt/addresstablemodel.cpp]: Agree, let's keep the comment but with the new method names

**Comment by laanwj** [comment]
Thanks, makes sense, if `std` has something that's usually preferable to using third party APIs.

File: src/qt/addresstablemodel.cpp (hunk 1 above)
Comment (fanquake) [reply src/qt/addresstablemodel.cpp]: Have reinstated the comment, using the new method names.

**Comment by practicalswift** [comment]
Concept ACK: less cruft is better.

**Comment by Sjors** [comment]
ACK c19ceb4. Tested on macOS 10.14.6 against depends (always spits out a trillion warnings at the end) as well as homebrew 5.13.0 (quite a few deprecation warnings remaining, but not these).

**Review by laanwj — APPROVED** [review]
ACK c19ceb4ea964a11d36ed108eb4d0975bbb38ac3a

**Comment by hebasto** [comment]
May I suggest to add `include <algorithm>` to the all modified `*.cpp` files?

From [Developer Notes - Source code organization](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#source-code-organization):
> - Every `.cpp` and `.h` file should `#include` every header file it directly uses classes, functions or other definitions from, even if those headers are already included indirectly through other headers.
>    -  _Rationale_: Excluding headers because they are already indirectly included results in compilation failures when those indirect dependencies change. Furthermore, it obscures what the real code dependencies are.

**Review by promag — COMMENTED** [review]
ACK 153d9dd9acffa95bb97a4b1579bd20237fdc9c52.

**Review by jonasschnelli — APPROVED** [review]
utACK 153d9dd9acffa95bb97a4b1579bd20237fdc9c52

**Comment by hebasto** [comment]
ACK 153d9dd9acffa95bb97a4b1579bd20237fdc9c52

**Comment by fanquake** [comment]
FWIW looks like these were marked deprecated in Qt 5.2:
> - QtAlgorithms
  With STL no longer being optional for building and using Qt, a number of parts
  of QtAlgorithms no longer make sense, and have therefore been deprecated.

**Pull request merged**